### PR TITLE
Add I18n support for the validators

### PR DESCRIPTION
**Pull request by sstok**
|Q            |A          |
|---          |---        |
|Bug Fix?     |no         |
|New Feature? |yes        |
|BC Breaks?   |yes (minor)|
|Deprecations?|no         |
|Fixed Tickets|#27        |
|License      |MIT        |

There is a minor BC break for the translation strings (removed the 'Your' part from the translated string).

### CODE REVIEW

**Comment by sstok** [comment]
@Soullivaneuh can you have a look at the implementation, and provide some suggestions if you have them? Thanks :+1:

**Comment by sstok** [comment]
Added a test to ensure the validator is registered properly (and fixed wrong service-id).

**Comment by Soullivaneuh** [comment]
> There is a minor BC break for the translation strings (removed the 'Your' part from the translated string).

Can't find what you're talking about. Can I have a ref link?

I'm not sure for the validator service. Why doing this? Related to translations?

Globally OK for the rest. If you want to compare, you can take a look to my isocodes validator library which handles translations for both Symfony and Silex: https://github.com/Soullivaneuh/IsoCodesValidator

**Comment by sstok** [comment]
Sorry for the late reply, I wasn't feeling well last week.

The original message for PasswordRequirement is something like `Your password must include at least one letter.` but the translated version for English is now `Password must include at least one letter.` It's not really a BC break but it is a change output.

*Looking at this, this can be changed to: `Password is not strong enough: - must include at least one letter.` (similar to the PasswordStrength messages)*

The validator injection is because we need to translate the suggestions before passing them as message arguments (but I really hope this is no big problem for other languages that have a more complex syntax).

**Comment by Soullivaneuh** [comment]
For message change, not sure it worth it to release a new major about that.

> The validator injection is because we need to translate the suggestions before passing them as message arguments

Sorry but I'm not sure to understand. For my project, I just have to create an integration for both Symfony bundle and Silex.

* https://github.com/Soullivaneuh/IsoCodesValidator/tree/c1b5407c99ae3308bd5d12792db9fd30be2e3a39/src/Bundle
* https://github.com/Soullivaneuh/IsoCodesValidator/blob/c1b5407c99ae3308bd5d12792db9fd30be2e3a39/src/Provider/IsoCodesValidatorServiceProvider.php

File: src/Resources/translations/validators.nl.xlf [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                <source>Your password must be at least {{length}} characters long.</source>
+                <target>Wachtwoord moet minstens {{length}} tekens lang zijn.</target>
+            </trans-unit>
+            <trans-unit id="2">
+                <source>Your password must include at least one letter.</source>
+                <target>Wachtwoord moet ten minste één letter bevatten.</target>
+            </trans-unit>
+            <trans-unit id="3">
+                <source>Your password must include both upper and lower case letters.</source>
+                <target>Wachtwoord moet ten minste één hoofd en kleine letter bevatten.</target>
```
Comment (mennowame) [inline src/Resources/translations/validators.nl.xlf]: would change this into "hoofdletter en kleine letter" or add a ~

File: src/Resources/translations/validators.nl.xlf [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            </trans-unit>
+            <trans-unit id="4">
+                <source>Your password must include at least one number.</source>
+                <target>Wachtwoord moet ten minste één nummer bevatten.</target>
+            </trans-unit>
+            <trans-unit id="5">
+                <source>Your password must contain at least one special character.</source>
+                <target>Wachtwoord moet ten minste één speciaal/leesteken bevatten.</target>
```
Comment (mennowame) [inline src/Resources/translations/validators.nl.xlf]: "speciaal teken of leesteken"

File: src/Resources/translations/validators.nl.xlf (hunk 1 above)
Comment (sstok) [inline src/Resources/translations/validators.nl.xlf]: Thanks for the suggestions :+1: I will add them ASAP.

**Comment by sstok** [comment]
> "For message change, not sure it worth it to release a new major about that."

Don't worry :) this is a minor release.

**Comment by sstok** [comment]
How did that " StyleCI / PHP-CS-Fixer" commit came-in :confused:

**Pull request merged**